src/clarity-angular/layout/vertical-nav/vertical-nav-group.ts

```diff
@@ -0,0 +1,186 @@
+                    } else if (!goingToCollapse && this.wasExpanded) {
+                        this.expandGroup();
+                        this.wasExpanded = false;
+                    }
+                }
+            )
+        );
+    }
+
+    private wasExpanded: boolean = false;
+
+    @Input("clrVerticalNavGroupExpanded")
+    set userExpandedInput(value: boolean) {
+        value = !!value;
+        if (this.expanded !== value) {
+            // We have to call toggleExpand because some cases require animations to occur first
+            // Directly setting the Expand service value skips the animation and can result in
+            // nodes in the DOM but the nav group still being collapsed
+            this.toggleExpand();
+        }
+    }
+
+    get expanded(): boolean {
+        return this._itemExpand.expanded;
+    }
+
+    set expanded(value: boolean) {
+        if (this._itemExpand.expanded !== value) {
+            this._itemExpand.expanded = value;
+            this.expandedChange.emit(value);
+        }
+    }
+
+    @Output("clrVerticalNavGroupExpandedChange")
+    expandedChange: EventEmitter<boolean> = new EventEmitter<boolean>(true);
+
+    @Input("clrVerticalNavGroupActive")
+    set active(value: boolean) {
+        value = !!value;
+        if (this._active !== value) {
+            if (value) {
+                this._verticalNavService.deactivateOtherNavGroups(this);
+            }
+            this._active = value;
+        }
+    }
+
+    private _subscriptions: Subscription[] = [];
+
+    private _expandAnimationState: string = COLLAPSED_STATE;
+
+    expandGroup(): void {
+        this.expanded = true;
+        // Expanded animation occurs after Expand.expand is set to true
+        this.expandAnimationState = EXPANDED_STATE;
+    }
+
+    collapseGroup(): void {
+        // If a Vertical Nav Group toggle button is clicked while the Vertical Nav is in Collapsed state,
+        // the Vertical Nav should be expanded first.
+        this.expandAnimationState = COLLAPSED_STATE;
+    }
+
+    // closes a group after the collapse animation
+    expandAnimationDone($event: any) {
+        if ($event.toState === COLLAPSED_STATE) {
+            this.expanded = false;
+        }
+    }
+
+    get expandAnimationState(): string {
+        return this._expandAnimationState;
+    }
+
+    set expandAnimationState(value: string) {
+        if (value !== this._expandAnimationState) {
+            this._expandAnimationState = value;
+        }
+    }
+
+    toggleExpand(): void {
+        if (this.expanded) {
+            this.collapseGroup();
+        } else {
+            // If nav is collasped, first open the nav
+            if (this._verticalNavService.collapsed) {
+                this._verticalNavService.collapsed = false;
+            }
+            //then expand the nav group
+            this.expandGroup();
+        }
+    }
+
+    get caretDirection(): string {
+        return (this.expanded) ? "down" : "right";
```

Comment:
Nitpick: Same as for `VerticalNav` itself, I could see this being better inlined in the template itself.